### DISCUSSION TITLE
Is configuration required in `ConfigProvider` to support rtl?

### DISCUSSION THREAD

**Discussion by dungsil** [unovue/reka-ui#1138] [Q&A] [answered]
hello
I am leaving a post because I have a question while looking at the documentation.

If you look at `ConfigProvider`, there is a props to set `dir`. (https://www.radix-vue.com/utilities/config-provider.html#config-provider-1)
Is this value mandatory even if the `html[dir="rtl"]` attribute is added to the HTML root?

**Comment by sadeghbarati** [comment, marked as answer]
Yes, based on `ConfigProvider` source code, the default `dir` is `ltr`, you must set `dir` prop to the component if you want to respect the `rtl` 

This prop is for FloatingUI components like `Popover`, `Menu`, and ... for the calculate `rtl` positions

https://www.radix-vue.com/guides/i18n.html#dynamic-direction

**Reply by dungsil** [reply]
thanks!
